community/kernel/src/main/java/org/neo4j/kernel/api/ReadOperations.java

```diff
@@ -125,6 +128,23 @@ PrimitiveLongIterator nodesGetFromIndexSeek( NewIndexDescriptor index, Object va
             throws IndexNotFoundKernelException;
 
     /**
+     * Returns an iterator with the matched entities.
+     *
+     * @throws org.neo4j.kernel.api.exceptions.index.IndexNotFoundKernelException if no such index found.
+     */
+    PrimitiveLongIterator indexQuery( NewIndexDescriptor index, IndexQuery... predicates ) throws IndexNotFoundKernelException;
+
+    /**
+     * Returns an iterator with the matched nodes.
+     *
+     * @throws org.neo4j.kernel.api.exceptions.index.IndexNotFoundKernelException if no such index found.
+     * @throws org.neo4j.kernel.api.exceptions.index.IndexNotApplicableKernelException if the index is not applicable
+     * for the given query.
+     */
+    PrimitiveLongIterator nodesGetFromCompositeIndexSeek( NewIndexDescriptor index, CompositeIndexQuery query )
```

Comment:
No we do not need that. We'll remove it when we get to adding composite index support to `indexQuery`.